src/Featurizers/NormalizeFeaturizer.h

```diff
@@ -0,0 +1,417 @@
+            std::vector<std::double_t> norms(Traits<std::vector<std::double_t>>::deserialize(ar));
+
+            return NormalizeTransformer<IteratorRangeT>(std::move(norms));
+        }()
+    ) {
+}
+
+template <typename IteratorRangeT>
+void NormalizeTransformer<IteratorRangeT>::save(Archive &ar) const /*override*/ {
+    // Version
+    Traits<std::uint16_t>::serialize(ar, 1); // Major
+    Traits<std::uint16_t>::serialize(ar, 0); // Minor
+
+    // Data
+    Traits<std::vector<std::double_t>>::serialize(ar, _norms);
+}
+
+template <typename IteratorRangeT>
+bool NormalizeTransformer<IteratorRangeT>::operator==(NormalizeTransformer const &other) const {
+
+#if (defined __clang__)
+#   pragma clang diagnostic push
+#   pragma clang diagnostic ignored "-Wfloat-equal"
+#endif
+
+    return this->_norms == other._norms;
+
+#if (defined __clang__)
+#   pragma clang diagnostic pop
+#endif
+
+}
+
+
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+template <typename IteratorRangeT>
+void NormalizeTransformer<IteratorRangeT>::execute_impl(IteratorRangeT const &input, typename BaseType::CallbackFunction const &callback) /*override*/ {
+    // unpack input to two iterators
+    IteratorType begin = std::get<0>(input);
+    IteratorType const & end = std::get<1>(input);
+
+    if (std::distance(begin, end) < 0) {
+        throw std::runtime_error("Input iterators to VectorNormsEstimator are invalid!");
+    }
+    if (_row >= _norms.size()) {
+        throw std::runtime_error("Number of norms is not aligned with number of rows!");
+    }
+
+    std::vector<std::double_t> res;
+    res.reserve(static_cast<size_t>(std::distance(begin, end)));
+    while (begin != end) {
+        if(is_null(begin, std::integral_constant<bool, Microsoft::Featurizer::Traits<ValueType>::IsNullableType>())) {
+            res.emplace_back(Traits<std::double_t>::CreateNullValue());
+        }
+        else {
+            res.emplace_back(static_cast<std::double_t>(static_cast<long double>(get_value(begin, std::integral_constant<bool, Microsoft::Featurizer::Traits<ValueType>::IsNullableType>())) / static_cast<long double>(_norms[_row])));
+        }
+        ++begin;
+    }
+    ++_row;
+    callback(res);
+}
+
+template <typename IteratorRangeT>
+bool NormalizeTransformer<IteratorRangeT>::is_null(IteratorType const &input, std::true_type) {
+    return Traits<ValueType>::IsNull(*input);
+}
+template <typename IteratorRangeT>
+bool NormalizeTransformer<IteratorRangeT>::is_null(IteratorType const &, std::false_type) {
+    return false;
+}
+
+
+template <typename IteratorRangeT>
+typename GetNullableType<typename std::iterator_traits<typename std::tuple_element<0, IteratorRangeT>::type>::value_type>::type NormalizeTransformer<IteratorRangeT>::get_value(IteratorType const &input, std::true_type) {

```

Comment:
>typename GetNullableType<typename std::iterator_traits<typename std::tuple_element<0, IteratorRangeT>::type>::value_type>::type [](start = 0, length = 127)

This should be `typename GetNullableType<typename std::iterator_traits<typename std::tuple_element<0, IteratorRangeT>::type>::value_type>::type const &` to avoid a copy.